cosmos/testing/network/network.go

```diff
@@ -158,13 +165,152 @@ func BuildGenesisState() map[string]json.RawMessage {
+		FinalTallyResult: &v1.TallyResult{},
+		SubmitTime:       &time.Time{},
+		DepositEndTime:   &time.Time{},
+		TotalDeposit:     sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(onehundred))),
+		VotingStartTime:  &voteStart,
+		VotingEndTime:    &voteEnd,
+		Metadata:         "metadata",
+		Title:            "title",
+		Summary:          "summary",
+		Expedited:        false,
+	}
+	// Append the proposal to the governance genesis state.
+	governanceState.Proposals = append(governanceState.Proposals, proposal)
+
+	// Set the states into the genesis state.
+	genState[authtypes.ModuleName] = encoding.Codec.MustMarshalJSON(&authState)
+	genState[banktypes.ModuleName] = encoding.Codec.MustMarshalJSON(&bankState)
+	genState[stakingtypes.ModuleName] = encoding.Codec.MustMarshalJSON(&stakingState)
+	genState[distrtypes.ModuleName] = encoding.Codec.MustMarshalJSON(&distrState)
+	genState[govtypes.ModuleName] = encoding.Codec.MustMarshalJSON(&governanceState)
 	return genState
 }
+
+func GetDistrValidator() common.Address {
+	pks := simtestutil.CreateTestPubKeys(2) //nolint: gomnd // magic numbers are fine in tests.
+	valConsPk1 := pks[1]
+	valConsAddr1 := sdk.ConsAddress(valConsPk1.Address())
+	valAddr := sdk.ValAddress(valConsAddr1)
+	return cosmlib.ValAddressToEthAddress(valAddr)
+}
+
+func DistrTestSetup(
+	bk *banktypes.GenesisState,
+	dk *distrtypes.GenesisState,
+	sk *stakingtypes.GenesisState,
+) {
+	// ==============================================================================
+	// Staking Keeper
+	// ==============================================================================
+
+	// Create the validator.
+	//nolint: gomnd // magic numbers are fine in tests.
+	pks := simtestutil.CreateTestPubKeys(2)
+	valConsPk1 := pks[1]
+	valConsAddr1 := sdk.ConsAddress(valConsPk1.Address())
+	valAddr := sdk.ValAddress(valConsAddr1)
+	val, err := distrtestutil.CreateValidator(valConsPk1, math.NewInt(onehundred))
+	if err != nil {
+		panic(err)
+	}
+	val.Status = stakingtypes.Unbonded
+	val.Tokens = math.NewInt(onehundred)
+	val.DelegatorShares = sdk.NewDec(onehundred)
+
+	// Set the validator.
+	sk.Validators = append(sk.Validators, val)
+
+	// Set the delegations.
+	delegator := cosmlib.AddressToAccAddress(TestAddress)
+	sk.Delegations = append(sk.Delegations, stakingtypes.Delegation{
+		DelegatorAddress: delegator.String(),
+		ValidatorAddress: valAddr.String(),
+		Shares:           val.DelegatorShares,
+	})
+
+	// ==============================================================================
+	// Distribution Keeper
+	// ==============================================================================
+
+	// Set the parameters.
+	dk.Params = distrtypes.DefaultParams()
+	dk.Params.WithdrawAddrEnabled = true
+
+	// Set the previous proposer.
+	dk.PreviousProposer = "polarvalcons1qqqqqqqqqqqqqqqqqqqqqqqqv9kxjcm9rl8r2z"
+
+	// Set the validator accumulated commission.
+	dk.ValidatorAccumulatedCommissions = append(dk.ValidatorAccumulatedCommissions,
```

Comment:
yeah you shouldnt newed any of this stuff.

Reply:
removed